### PR TITLE
stat/distuv: add AlphaStable distribution

### PR DESCRIPTION
**Pull request by romanwerpachowski**
Implements sampling from a general alpha-stable distribution and some basic distribution properties when possible.

### CODE REVIEW

File: stat/distuv/alphastable_test.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+	d := AlphaStable{Alpha: 1.75, Beta: 0.2, Scale: 1.2, Mu: 0.3, Src: src}
+	n := 100000
+	x := make([]float64, n)
+	for i := 0; i < n; i++ {
+		x[i] = d.Rand()
+	}
+	checkMean(t, 0, x, d, 1e-2)
+}
+
+func TestAlphaStableCauchy(t *testing.T) {
```
Comment (kortschak) [inline stat/distuv/alphastable_test.go:87]: s/Cauchy/Mode/?

File: stat/distuv/alphastable.go [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+// Median returns the median of the distribution.
+// Median panics when Beta != 0, because then the mode is not analytically
+// expressible.
+func (a AlphaStable) Median() float64 {
+	if a.Beta == 0 {
+		return a.Mu
+	}
+	panic("Cannot compute Median for Beta != 0")
```
[the remark this answers is not in the document]
Comment (romanwerpachowski) [reply stat/distuv/alphastable.go:52]: NaN means "undefined", but here the result exists, we just don't know how to calculate it.

File: stat/distuv/alphastable.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+}
+
+// Mode returns the mode of the distribution.
+// Mode panics when Beta != 0, because then the mode is not analytically
+// expressible.
+func (a AlphaStable) Mode() float64 {
+	if a.Beta == 0 {
+		return a.Mu
+	}
+	panic("Cannot compute Mode for Beta != 0")
```
[the remark this answers is not in the document]
Comment (romanwerpachowski) [reply stat/distuv/alphastable.go:62]: Same here.

File: stat/distuv/alphastable.go [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+// AlphaStable represents an α-stable distribution with four parameters:
+//  stability 0 < α ≤ 2
+//  skewness -1 ≤ β ≤ 1
+//  scale c > 0
+//  location μ
+// More information at https://en.wikipedia.org/wiki/Stable_distribution
+type AlphaStable struct {
+	Alpha float64
+	Beta  float64
+	Scale float64
```
[the remark this answers is not in the document]
Comment (romanwerpachowski) [reply stat/distuv/alphastable.go:22]: I think "Scale" is more descriptive.

File: stat/distuv/alphastable_test.go (hunk 1 above)
Comment (romanwerpachowski) [reply stat/distuv/alphastable_test.go:87]: It only tests Mode for the Cauchy distribution (because there is nothing else to test).

**Comment by romanwerpachowski** [comment]
Thanks, PTAL.

File: stat/distuv/alphastable.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (kortschak) [reply stat/distuv/alphastable.go:22]: It is, but the consistency is lost; you have Alpha not Stability, and Beta not Skewness. I would use the parameters that are named and add comments to the fields explaining what they do.

File: stat/distuv/alphastable.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (kortschak) [reply stat/distuv/alphastable.go]: OK

File: stat/distuv/alphastable.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (romanwerpachowski) [reply stat/distuv/alphastable.go:22]: OK.

**Comment by romanwerpachowski** [comment]
PTAL.

File: stat/distuv/alphastable.go [hunk 5]
```diff
@@ -17,11 +17,15 @@ import (
 //  location μ
 // More information at https://en.wikipedia.org/wiki/Stable_distribution
 type AlphaStable struct {
+	// Stability parameter 0 < α ≤ 2.
```
Comment (kortschak) [inline stat/distuv/alphastable.go:20]: These repeat what exists in the comment for the type.

```
// AlphaStable represents an α-stable distribution with four parameters.
// See https://en.wikipedia.org/wiki/Stable_distribution for details.
type AlphaStable struct {
	// Alpha is the stability parameter for the distribution.
	// it is valid within the range 0 < α ≤ 2.
	Alpha float64
	// Beta is the skewness parameter for the distribution.
	// it is valid within the range -1 ≤ β ≤ 1.
	Beta float64
	// C is the scale parameter for the distribution.
	// it is valid when positive.
	C float64
	// Mu is the location parameter for the distribution.
	Mu  float64

	Src rand.Source
}
```

Comment (romanwerpachowski) [reply stat/distuv/alphastable.go]: Good point. Updated the comment. PTAL.

**Comment by romanwerpachowski** [comment]
PTAL.

**Pull request merged**